### PR TITLE
added "download from the nexus" functionality

### PR DESCRIPTION
**Pull request by PavelAlennikov**

### CODE REVIEW

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 1]
```diff
@@ -222,7 +211,7 @@ protected void updateValuesWithConfig() {
     }
 
     protected void handleException(Exception e, Architecture arch,
-            String version) {
+                                   String version) {
```
Comment (0I0u3uk) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java:221]: Fix formating

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 2]
```diff
@@ -531,6 +524,42 @@ protected Integer versionCompare(String str1, String str2) {
...
     }
 
+    protected List<URL> getDriversFromNexus(URL driverUrl) throws IOException {
+        if (!mirrorLog) {
+            log.info("Crawling driver list from nexus {}", driverUrl);
+            mirrorLog = true;
+        } else {
+            log.trace("[Recursive call] Crawling driver list from nexus {}",
+                    driverUrl);
+        }
```
Comment (0I0u3uk) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java:541]: this is gebbuging code?

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 3]
```diff
@@ -531,6 +524,42 @@ protected Integer versionCompare(String str1, String str2) {
...
+
+        String driverStr = driverUrl.toString();
```
Comment (0I0u3uk) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java]: You are using this variable only in one place, maybe use driverUrl.toString() in this place?

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 4]
```diff
@@ -143,8 +127,13 @@ protected File postDownload(File archive) {
     }
 
     protected String getCurrentVersion(URL url, String driverName) {
-        return url.getFile().substring(url.getFile().indexOf(SLASH) + 1,
-                url.getFile().lastIndexOf(SLASH));
+        if (isUsingNexus) {
+            String[] urlParts = url.getFile().split("/");
+            return urlParts[urlParts.length - 2];
```
Comment (clicman) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java]: Possible ArrayOutOfBoundsException

File: src/main/java/io/github/bonigarcia/wdm/UrlFilter.java [hunk 5]
```diff
@@ -67,16 +66,18 @@
...
-        if (out.size() > 1 && arch != null) {
+        if (out.size() > 0 && arch != null) {
             for (URL url : list) {
-                if (!url.getFile().contains("x86")
+                final String urlFile = url.getFile();
+                final String fileName = urlFile.substring(urlFile.lastIndexOf("/"), urlFile.length());
+                if (!fileName.contains("x86")
                         && !url.getFile().contains("64")
                         && !url.getFile().contains("i686")
                         && !url.getFile().contains("32")) {
```
Comment (clicman) [inline src/main/java/io/github/bonigarcia/wdm/UrlFilter.java]: Move archs to constants. do not forgot to change all hardcode to constants

**Review by clicman — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java (hunk 4 above)
Comment (PavelAlennikov) [reply src/main/java/io/github/bonigarcia/wdm/BrowserManager.java]: fixed

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java (hunk 1 above)
Comment (PavelAlennikov) [reply src/main/java/io/github/bonigarcia/wdm/BrowserManager.java:221]: solved

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java (hunk 2 above)
Comment (PavelAlennikov) [reply src/main/java/io/github/bonigarcia/wdm/BrowserManager.java:541]: solved

**Comment by PavelAlennikov** [comment]
@clicman @0I0u3uk @SandraBarmina @kosteman 
guys changes are stable. You can review now

**Review by 0I0u3uk — APPROVED** [review]

File: src/main/resources/log4j.properties [hunk 6]
```diff
@@ -0,0 +1,11 @@
+log4j.rootLogger=DEBUG, consoleAppender, fileAppender
```
Comment (0I0u3uk) [inline src/main/resources/log4j.properties]: Релизить лучше с INFO я думаю, а то пользователь офигеет

**Review by 0I0u3uk — APPROVED** [review]

File: pom.xml [hunk 7]
```diff
@@ -1,10 +1,10 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>io.github.bonigarcia</groupId>
+    <groupId>ru.sbtqa</groupId>
     <artifactId>webdrivermanager</artifactId>
-    <version>2.0.2-SNAPSHOT</version>
+    <version>2.0.3-SNAPSHOT</version>
```
Comment (kosteman) [inline pom.xml]: 2.1-SNAPSHOT

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 8]
```diff
@@ -695,6 +727,8 @@ protected void reset() {
         mirrorLog = false;
         isForcingCache = false;
         isForcingDownload = false;
+        isUsingNexus = false;
```
Comment (kosteman) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java]: usingNexus

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 9]
```diff
@@ -464,7 +459,11 @@ protected void handleDriver(URL url, String driver, List<URL> out) {
     }
 
     protected boolean isUsingTaobaoMirror() {
-        return getDriverUrl().getHost().equalsIgnoreCase("npm.taobao.org");
+        return isUsingTaobao;
+    }
+
+    protected boolean isUsingNexus() {
```
Comment (kosteman) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java]: protected boolean isUsingNexus() {
    return usingNexus;
}

File: src/main/java/io/github/bonigarcia/wdm/BrowserManager.java [hunk 10]
```diff
@@ -115,6 +97,8 @@
     protected String driverVersionKey;
     protected String driverUrlKey;
     protected String[] ignoredVersions;
+    protected boolean isUsingTaobao;
+    protected boolean isUsingNexus;
```
Comment (kosteman) [inline src/main/java/io/github/bonigarcia/wdm/BrowserManager.java]: why these variables are not private?

**Review by kosteman — CHANGES_REQUESTED** [review]

**Review by clicman — APPROVED** [review]

**Review by kosteman — APPROVED** [review]

**Pull request merged**